bin/which

```diff
@@ -100,21 +100,30 @@ foreach my $command (@ARGV) {
     }
 
     foreach my $dir (@PATH) {
+        my $path = $dir . $file_sep . $command;
+
+        if (-d $path) {
+            next;
+        }
         if ($^O eq 'MacOS') {
```

Comment:
Do you know why we'd single out MacOS here? I think this is Mac Classic, which we don't need to support. We don't need to change this for this PR, but it's something to think about. I wonder what happens on Windows, which has a completely different permissions model.